### DISCUSSION TITLE
Help with PDS3 Target_Identification, type mapping

### DISCUSSION THREAD

**Discussion by thareUSGS** [pds-data-dictionaries/ldd-management#199] [General]
We have a PDART coming in which migrates voyager ISS NA (Imaging Science Subsystem Narrow Angle Camera) to PDS4. The original TARGET_NAME in the Voyager archive is all over the place (including many that are just blank). See some entries below. For many we use something like:

```
<Target_Identification>
  <name>ALPHA_CAN</name>
  <type>Star</type>
</Target_Identification>
```

But for the odd ones and empty ones there is not really a good "type" ([see](https://pds.nasa.gov/datastandards/documents/dd/current/PDS4_PDS_DD_1G00.html#N72540711)).

Options we are thinking of might be:
```
<Target_Identification>
  <name>unspecified target name</name>
  <type>Planetary System</type>
</Target_Identification>
```

Or we could perhaps put an enumeration SCR is to get something like  "Unknown" or "Untargeted" with the type enumeration. This would need DDWG review and vote. Would look like this:
```
<Target_Identification>
  <name>unspecified target name</name>
  <type>Unknown</type>
</Target_Identification>
```

thoughts or ideas on either?

# full listing and current type mapping
<details>
  <summary>Click to expand!</summary>
<pre>
TARGET_NAME='?         ' -> 
TARGET_NAME='' -> 
TARGET_NAME='263/-15   ' -> 
TARGET_NAME='267/-24   ' -> 
TARGET_NAME='269/-22.5 ' -> 
TARGET_NAME='269/-22.9 ' -> 
TARGET_NAME='272/-21   ' -> 
TARGET_NAME='ALPHA_CAN ' -> star
TARGET_NAME='ALPHA_CANI' -> star
TARGET_NAME='ALPHA_CENT' -> star
TARGET_NAME='ALPHA_LEON' -> star
TARGET_NAME='ALPHA_LYRA' -> star
TARGET_NAME='ALPHALYRAE' -> star
TARGET_NAME='ALPHA_PAV ' -> star
TARGET_NAME='ALPHA_PAVO' -> star
TARGET_NAME='ALPHA_VIR ' -> star
TARGET_NAME='ASTEROID_B' -> asteroid
TARGET_NAME='AZ/EL     ' -> 
TARGET_NAME='BETA_CANIS' -> star
TARGET_NAME='BETA_CENTA' -> star
TARGET_NAME='BETA_CMA  ' -> star
TARGET_NAME='BETA_PER  ' -> star
TARGET_NAME='BETA_PERSE' -> star
TARGET_NAME='BODARK    ' -> calibration field
TARGET_NAME='CAL_LAMPS ' -> calibration field
TARGET_NAME='CONE=20   ' -> 
TARGET_NAME='CONE=20_DE' -> 
TARGET_NAME='CONE=24   ' -> 
TARGET_NAME='CONE=24_DE' -> 
TARGET_NAME='CONE=27   ' -> 
TARGET_NAME='CONE=27_DE' -> 
TARGET_NAME='CONE=30   ' -> 
TARGET_NAME='CONE=30_DE' -> 
TARGET_NAME='CONE=35   ' -> 
TARGET_NAME='CONE=70   ' -> 
TARGET_NAME='DARK      ' -> calibration field
TARGET_NAME='DARK_SKY  ' -> calibration field
TARGET_NAME='DELTA_SCO ' -> star
TARGET_NAME='DUST_BAND ' -> dust
TARGET_NAME='EARTH     ' -> planet
TARGET_NAME='ETA_URSA_M' -> star
TARGET_NAME='GAMMA_PEG ' -> star
TARGET_NAME='HGA_EDGE  ' -> equipment
TARGET_NAME='HR_7567   ' -> star
TARGET_NAME='IOTA_HER  ' -> star
TARGET_NAME='JUPITER   ' -> planet
TARGET_NAME='KY_ANDROME' -> star
TARGET_NAME='MARS      ' -> planet
TARGET_NAME='NEPTUNE   ' -> planet
TARGET_NAME='NEPTUNE' ->    planet
TARGET_NAME='NEREID    ' -> satellite
TARGET_NAME='NO_TARGET ' -> 
TARGET_NAME='N. URANUS ' -> planet
TARGET_NAME='PLAQUE    ' -> equipment
TARGET_NAME='PRA       ' -> equipment
TARGET_NAME='PWS       ' -> equipment
TARGET_NAME='RA/DEC    ' -> 
TARGET_NAME='RA/DEC=151' -> 
TARGET_NAME='RA/DEC=255' -> 
TARGET_NAME='RA/DEC=-54' -> 
TARGET_NAME='RA/DEC=77.' -> 
TARGET_NAME='RA/DEC=78.' -> 
TARGET_NAME='RA/DEC=83.' -> 
TARGET_NAME='SAO_#52742' -> star
TARGET_NAME='SATURN    ' -> planet
TARGET_NAME='SCAT_LIGHT' -> dust
TARGET_NAME='SIG_2_CORO' -> star
TARGET_NAME='SIGMA_SAG ' -> star
TARGET_NAME='SIGMA_SAGI' -> star
TARGET_NAME='SIGMA_SGR ' -> star
TARGET_NAME='SKY       ' -> 
TARGET_NAME='SKY___	   ' -> 
TARGET_NAME='SKY_	     ' -> 
TARGET_NAME='SKY		     ' -> 
TARGET_NAME='SUN       ' -> star
TARGET_NAME='SW URANUS ' -> planet
TARGET_NAME='THETA_CORO' -> star
TARGET_NAME='THETA_OPH ' -> star
TARGET_NAME='URANUS    ' -> planet
TARGET_NAME='URANUS' ->     planet
TARGET_NAME='UVS_SOURCE' -> calibration field
TARGET_NAME='VEGA      ' -> star
TARGET_NAME='VENUS     ' -> planet
</pre>
</details>

**Comment by jordanpadams** [comment]
@thareUSGS is the plan to create and refer to context products if they do not already exist? also, looks like there is a "unk" context product, but not sure it makes sense as type "Calibrator". https://pds.nasa.gov/data/pds4/context-pds4/target/calibrator.unk_1.0.xml . We also have a "PDS3 Target" that was migrated from the old catalog here: https://pds.nasa.gov/data/pds4/context-pds3/target/Product/target_UNK_1.0.xml.

My two cents: New unknown context product with target_type unknown.

**Comment by rgdeen** [comment]
Context products don’t seem to be required, which is very good… we have no idea what some of these are.  Happy to leave context pointer blank.  Or are you suggesting they all point to a single “unk” context product?   I would strenuously argue though that even though we don’t know what they are, including the string as written (for target name) has very high value because someone might be able to intuit the meaning.  I might look askance at a primary data provider doing that (they should know their data), but we’re not the primary, just trying to make historical data more easily accessible to folks. 

Anyway, an unknown type would be the ideal solution I think.  Given that’s in the core PDS DD, what would be the availability schedule for such a change?  We’ll have to check with the PDART PI, I know he wants to get it out the door before finding runs out.

**Reply by jordanpadams** [reply]
@rgdeen I would hope we (PDS) would create these context products for all of these targets, even if little is known about the targets to enable search. If we focus to narrowly on search for this mission, we lose the possible interoperability across the system (or even search across the system). The context products are where you can track the synonyms and/or irregular naming of these targets that came from the missions. By putting these values as the Target_Identification.name, without updating the metadata to the PDS4 Standard and guidelines for good metadata, we are just creating a PDS3 archive in a PDS4-like structure. The metadata should be improved where possible. The "irregular" PDS3 metadata should be captured elsewhere in the system, not in the Target_Identification area.

**Comment by rgdeen** [comment]
But where?  Yes you're copying pds3-isms, but that's because it's pds3 data!!  And Target_PDS3 is only available in Product_Target_PDS3, which seems, shall we say, somewhat less than useful.  I'm not going to kick it out of Product_Observational over a target name!!  Product_Target_PDS3 doesn't even support File_Area_Observational.

You can't let the perfect be the enemy of the good here.  I wouldn't let a primary provider get away with this, but given that we're just trying to provide a service to the community (reformatting the data to make it easier to use and be marginally pds4 compliant), it's not really up to us to define every little thing.  There will be warts because it's old and poorly documented data.  PDS just needs to accept that.  Especially because the meat of the PDART is only a few hundred images; we're providing 30k "EDR's" reformatted as above because we have the code in-hand and it's easy to do.  If there are too many barriers, we might not be able to support this, and that would be a huge loss to the community.

That said, I guess if EN or IMG really wants to make 84 context products for all the above, knock yourself out.  But a good chunk of them would say the same thing: "Unknown target assigned by Voyager team", which is not very multimission or reusable.  And that still doesn't address what type they should be, or when an "unknown" type might become available.

**Reply by mit3ch** [reply]
Okay. I'm asking that we pause the discussion please. RMS has not been involved and needs to be. RMS is the lead for the migration of Voyager image data to PDS4. We anticipate producing a mission level namespace for that effort. PPI generated separate namespaces because the instrument packages on the two spacecraft differed. I have questions, but no time this week or next. I am not on board with the proposed solutions because I do not know the problems yet. I would prefer to avoid dealing with this one issue at a time, so I need a big picture view of what is being done, what metadata is appropriate but not currently available in PDS4, and much more in order to be able to help. I propose a zoom meeting of all interested parties during the week of April 18th if someone can set up a doodle poll or some similar approach to scheduling.

**Comment by matthewtiscareno** [comment]
Hi.  We need to take a step back here, so that we can come up to speed regarding how this project will fit together with the RMS project of migrating all Voyager ISS data to PDS4.   There is a lot of work that we at RMS will do to properly craft the mission dictionary/dictionaries.

**Comment by rgdeen** [comment]
Hey.. happy to accept input and work together on this, but the data is for a PDART that can't wait a super long time because it will run out of funding (Dan Wenkert is the PI).  The PDART is analyzing high phase angle images of Uranus and Neptune to help characterize exoplanets, but we are planning on delivering ~30k EDRs too (based on QEDRs) since we have the scripts to process them.   That said, mid-April is certainly not too late.

**Comment by rgdeen** [comment]
I should also note that this discussion has nothing at all to do with the Voyager LDD.  We're just discussing how to shoehorn things into the PDS base classes to work around (excessive) PDS restrictions in terms of required data.  The VGR LDD is a separate discussion, and happy to have it.  So, we should proceed with this discussion, which is really about how to get around the lack of "unknown" as a target type in the core DD, and whether a context product is needed for every target (technically per the DD, it is not).

**Comment by acraugh** [comment]
A couple of target suggestions that come with their own baggage: 
1. You could request that both &lt;name&gt; and &lt;type&gt; within &lt;Target_Identification&gt; be nillable (this would require a core schema update, but it's backwards-compatible, ostensibly). Then you could explicitly indicate that the values are nil because they are "unknown" without having to kludge a target named "unknown" and a target type of "unknown".
2. You might also consider values like "Unspecified" or "Pointing", which are subtly different from "unknown" but might be a better description of 'sky' observations intentionally made for, say, check-out.

**Comment by rgdeen** [comment]
I actually kind of like the nillable idea here @acraugh .  Stylistically I prefer a value of "unknown" but that's not the PDS way.  ;-)  Nillable with reason "unknown" neatly solves the problem.

**Reply by acraugh** [reply]
(Been falling behind in GitHub reading...) Actually, @rgdeen, when some attribute is defined as nillable in a PDS4 label you must also specify why it is nil - and "unkown" is one of the reasons (as are "inapplicable" and one or two other reasons I'd have to look up). This is specified through the "nilReason" XML attribute. So it would look something like this in an actual label: `<type xsi:nil="true" nilReason="unknown"/>`. It's about as close as you can get to having your cake and eating it, too, given XSD constraints.
